walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Iterator;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) { //Was able to get the runtime down from ~4.5 secs to ~2.8 secs by removing unnecessary loops(condensing the logic), then down to ~2.6 secs by simplifying conditionals
+    long meetingDuration = request.getDuration();
+    if(meetingDuration > 24*60)
+    {
+        return new ArrayList<TimeRange>();
+    }
+
+    TimeRange initial = TimeRange.fromStartEnd(0,24*60,false);  //whole day
```

Comment:
Nit: Comment next to statement is fine here - just add a space between "//" and text, with proper caps: "// Whole day"